internal/generate/generate_modules.go

```diff
@@ -80,6 +97,29 @@ func newTemplates(moduleDir string, outputDir string, overwrite bool) []*Templat
 			continue
 		}
 
+		// detect the file type
+		detectedMIME, err := mimetype.DetectFile(path)
+		if err != nil {
+			panic(err)
+		}
+
+		// detect if the file type is binary
+		isBinary := true
+		for mime := detectedMIME; mime != nil; mime = mime.Parent() {
+			if mime.Is("text/plain") {
+				isBinary = false
+			}
+		}
+
+		if isBinary {
+			binFile = append(binFile, &BinFileConfig{
+				source:      path,
+				destination: fs.ReplacePath(path, moduleDir, outputDir),
+				mineType:    detectedMIME.String(),
+			})
+			continue
+		}
+
 		_, file := filepath.Split(path)
```

Comment:
Mind removing this block? We don't use it anymore, as we just try to template everything in the repo (that's not binary)